pkg/operator/controller/externaldns/credentials-request.go

```diff
@@ -0,0 +1,179 @@
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/apimachinery/pkg/types"
+
+	operatorv1alpha1 "github.com/openshift/external-dns-operator/api/v1alpha1"
+	controller "github.com/openshift/external-dns-operator/pkg/operator/controller"
+)
+
+// ensureExternalCredentialsRequest ensures that the externalDNS credential request exists.
+// Returns a boolean if the credential request exists, its current state if it exists
+// and an error if it cannot be created or updated.
+func (r *reconciler) ensureExternalCredentialsRequest(ctx context.Context, externalDNS *operatorv1alpha1.ExternalDNS) (bool, *cco.CredentialsRequest, error) {
+	name := controller.ExternalDNSCredentialsRequestName(externalDNS)
+
+	desired, err := desiredCredentialsRequest(name, externalDNS)
+	if err != nil {
+		return false, nil, err
+	}
+
+	exists, current, err := r.currentExternalDNSCredentialsRequest(ctx, name)
+
+	if err != nil {
+		return false, nil, err
+	}
+
+	if !exists {
+		if err := r.createExternalDNSCredentialsRequest(ctx, desired); err != nil {
+			return false, nil, err
+		}
+		return r.currentExternalDNSCredentialsRequest(ctx, name)
+	}
+
+	return true, current, nil
+}
+
+func awsProviderConfig(codec *cco.ProviderCodec) (*runtime.RawExtension, error) {
+	return codec.EncodeProviderSpec(
+		&cco.AWSProviderSpec{
+			TypeMeta: metav1.TypeMeta{
+				Kind: "AWSProviderSpec",
+			},
+			StatementEntries: []cco.StatementEntry{
+				{
+					Effect: "Allow",
+					Action: []string{
+						"route53:ChangeResourceRecordSets",
+					},
+					Resource: "arn:aws:route53:::hostedzone/*",
+				},
+				{
+					Effect: "Allow",
+					Action: []string{
+						"route53:ListHostedZones",
+						"route53:ListResourceRecordSets",
+					},
+					Resource: "*",
+				},
+			},
+		})
+}
+
+func azureProviderConfig(codec *cco.ProviderCodec) (*runtime.RawExtension, error) {
+	return codec.EncodeProviderSpec(
+		&cco.AzureProviderSpec{
+			TypeMeta: metav1.TypeMeta{
+				Kind: "AzureProviderSpec",
+			},
+			RoleBindings: []cco.RoleBinding{
+				{Role: "Contributor"},
+			},
+		})
+}
+
+func gcpProviderConfig(codec *cco.ProviderCodec) (*runtime.RawExtension, error) {
+	return codec.EncodeProviderSpec(
+		&cco.GCPProviderSpec{
+			TypeMeta: metav1.TypeMeta{
+				Kind: "GCPProviderSpec",
+			},
+			PredefinedRoles: []string{
+				"roles/dns.admin",
+			},
+		})
+}
+
+// desiredCredentialsRequestName returns the desired credentials request definition for externalDNS
+func desiredCredentialsRequest(name types.NamespacedName, externalDNS *operatorv1alpha1.ExternalDNS) (*cco.CredentialsRequest, error) {
+	credentialsRequest := &cco.CredentialsRequest{
+		ObjectMeta: metav1.ObjectMeta{
+			Name:      name.Name,
+			Namespace: name.Namespace,
+		},
+		Spec: cco.CredentialsRequestSpec{
+			ServiceAccountNames: []string{"external-dns-operator"},
+			SecretRef: k8sv1.ObjectReference{
+				Name:      "externaldns-cloud-credentials",
+				Namespace: "external-dns-operator",
```

Comment:
There is a config field for the operator namespace: [link](https://github.com/openshift/external-dns-operator/blob/main/pkg/operator/config/config.go#L48).